### PR TITLE
feat(userspace): deprecate `-d` daemonize option

### PR DESCRIPTION
**Pull request by incertum**
Symptoms included the message queue filling up without popping any messages even though events were handled normally.

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

> /kind release

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area build

/area engine

> /area tests

> /area proposals

> /area CI

**What this PR does / why we need it**:

Update :  feat(userspace): deprecate `-d` daemonize option

@falcosecurity/falco-maintainers 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
feat(userspace): deprecate `-d` daemonize option
```

### CODE REVIEW

**Review by leogr — COMMENTED** [review]
@incertum Does this PR just correct the comments?

**Comment by leogr** [comment]
Questions:
- Have you tried to disable `stdout` and use just `file`?
- Do we really need the `-d` option? :thinking:

**Comment by incertum** [comment]
>   Have you tried to disable `stdout` and use just `file`?

yes same outcome, that is, no outputs. My debugging journey came to the conclusion that somehow we fail to reach the code flow where we pop from the queue when using `-d`, something that of course makes no sense as I could confirm we got events and were putting them into the queue ...

>  Do we really need the `-d` option?

I actually synced with SREs and the answer would be no. However, we should keep the pidfile option. That was the one appealing aspect of using `-d` for one environment, but we already changed the deployment and don't need `-d` anymore anyways.

@leogr depending on how much engineering resources we have, I am of course ok with keeping it if someone finds a fix. At the same time, it would not be good to keep supporting something that is broken. In addition, removing some older features not urgently needed will be better for all of us maintainers, because we reduce maintenance overhead and make space for new features we urgently do need. Therefore, I would be ok to just remove it without a formal deprecation process since it is already broken in the current release. Again, we can keep the pidfile option without needing `-d`.

If above sounds like a plan I can go ahead and repurpose this PR, else I may just close it as I am not advancing with finding a fix.

**Comment by leogr** [comment]
> > Do we really need the `-d` option?

> I actually synced with SREs and the answer would be no. However, we should keep the pidfile option. That was the one appealing aspect of using `-d` for one environment, but we already changed the deployment and don't need `-d` anymore anyways.
> 
> @leogr depending on how much engineering resources we have, I am of course ok with keeping it if someone finds a fix. At the same time, it would not be good to keep supporting something that is broken. In addition, removing some older features not urgently needed will be better for all of us maintainers, because we reduce maintenance overhead and make space for new features we urgently do need. Therefore, I would be ok to just remove it without a formal deprecation process since it is already broken in the current release. Again, we can keep the pidfile option without needing `-d`.

Since using `systemd` or Kubernetes is the default way to go, I find the `-d` flag somewhat redundant and, as you pointed out, potentially confusing. I actually don't see any compelling use case where that option should be used.

So, I'm in favor to remove it just because not having that option would provide a better UX, regardless of the fact that's broken. Ofc, I wouldn't fix it if we decide to remove it.

Also, I totally agree with keeping the `--pidfile` option without `-d`. Btw, it looks like we are already using `--pidfile` this way :point_down: 
https://github.com/falcosecurity/falco/blob/master/scripts/systemd/falco-kmod.service

> If above sounds like a plan I can go ahead and repurpose this PR, else I may just close it as I am not advancing with finding a fix.

let's see what others think cc @falcosecurity/falco-maintainers

**Comment by Andreagit97** [comment]
Sorry, I'm late to the party, agree with removing this!

**Comment by incertum** [comment]
Removed `-d` flag in case we want to follow through with the deprecation. Something else missing or did I remove it properly?

**Comment by leogr** [comment]
> Removed `-d` flag in case we want to follow through with the deprecation. Something else missing or did I remove it properly?

One file is conflicting. Rebase pls. :pray:

File: userspace/falco/app/actions/pidfile.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	{
+		int64_t self_pid = getpid();
+
+		std::ofstream pidfile;
+		pidfile.open(s.options.pidfilename);
+
+		if (!pidfile.good())
+		{
+			falco_logger::log(LOG_ERR, "Could not write pid to pidfile " + s.options.pidfilename + ". Exiting.\n");
+			exit(-1);
```
[the remark this answers is not in the document]
Comment (Andreagit97) [reply userspace/falco/app/actions/pidfile.cpp]: i've done it :)

**Review by Andreagit97 — DISMISSED** [review]
/approve

**Comment by Andreagit97** [comment]
we need a rebase :roll_eyes:

**Review by leogr — APPROVED** [review]

**Pull request merged**